tests/components/season/test_sensor.py

```diff
@@ -28,217 +36,90 @@
-        )
-        state = self.hass.states.get("sensor.season")
-        assert state.attributes.get("friendly_name") == "Season"
-
-    def test_setup_hemisphere_equator(self):
-        """Test platform setup of equator."""
-        self.hass.config.latitude = HEMISPHERE_EQUATOR["homeassistant"]["latitude"]
-        assert setup_component(self.hass, "sensor", HEMISPHERE_EQUATOR)
-        self.hass.block_till_done()
-        assert (
-            self.hass.config.as_dict()["latitude"]
-            == HEMISPHERE_EQUATOR["homeassistant"]["latitude"]
-        )
-        state = self.hass.states.get("sensor.season")
-        assert state.attributes.get("friendly_name") == "Season"
-
-    def test_setup_hemisphere_empty(self):
-        """Test platform setup of missing latlong."""
-        self.hass.config.latitude = None
-        assert setup_component(self.hass, "sensor", HEMISPHERE_EMPTY)
-        self.hass.block_till_done()
-        assert self.hass.config.as_dict()["latitude"] is None
+NORTHERN_PARAMETERS = [
+    (TYPE_ASTRONOMICAL, datetime(2017, 9, 3, 0, 0), STATE_SUMMER),
+    (TYPE_METEOROLOGICAL, datetime(2017, 8, 13, 0, 0), STATE_SUMMER),
+    (TYPE_ASTRONOMICAL, datetime(2017, 9, 23, 0, 0), STATE_AUTUMN),
+    (TYPE_METEOROLOGICAL, datetime(2017, 9, 3, 0, 0), STATE_AUTUMN),
+    (TYPE_ASTRONOMICAL, datetime(2017, 12, 25, 0, 0), STATE_WINTER),
+    (TYPE_METEOROLOGICAL, datetime(2017, 12, 3, 0, 0), STATE_WINTER),
+    (TYPE_ASTRONOMICAL, datetime(2017, 4, 1, 0, 0), STATE_SPRING),
+    (TYPE_METEOROLOGICAL, datetime(2017, 3, 3, 0, 0), STATE_SPRING),
+]
+
+SOUTHERN_PARAMETERS = [
+    (TYPE_ASTRONOMICAL, datetime(2017, 12, 25, 0, 0), STATE_SUMMER),
+    (TYPE_METEOROLOGICAL, datetime(2017, 12, 3, 0, 0), STATE_SUMMER),
+    (TYPE_ASTRONOMICAL, datetime(2017, 4, 1, 0, 0), STATE_AUTUMN),
+    (TYPE_METEOROLOGICAL, datetime(2017, 3, 3, 0, 0), STATE_AUTUMN),
+    (TYPE_ASTRONOMICAL, datetime(2017, 9, 3, 0, 0), STATE_WINTER),
+    (TYPE_METEOROLOGICAL, datetime(2017, 8, 13, 0, 0), STATE_WINTER),
+    (TYPE_ASTRONOMICAL, datetime(2017, 9, 23, 0, 0), STATE_SPRING),
+    (TYPE_METEOROLOGICAL, datetime(2017, 9, 3, 0, 0), STATE_SPRING),
+]
+
+
+def idfn(val):
+    """Provide IDs for pytest parametrize."""
+    if isinstance(val, (datetime)):
+        return val.strftime("%Y%m%d")
+
+
+@pytest.mark.parametrize("type,day,expected", NORTHERN_PARAMETERS, ids=idfn)
+async def test_season_northern_hemisphere(hass, type, day, expected):
+    """Test that season should be summer."""
+    hass.config.latitude = HEMISPHERE_NORTHERN["homeassistant"]["latitude"]
+
+    config = {
+        **HEMISPHERE_NORTHERN,
+        "sensor": {"platform": "season", "type": type},
+    }
+
+    with patch("homeassistant.components.season.sensor.utcnow", return_value=day):
+        assert await async_setup_component(hass, "sensor", config)
+        await hass.async_block_till_done()
+
+    state = hass.states.get("sensor.season")
+    assert state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6271139)